modules/RTC/TPCUtils.js

```diff
@@ -357,14 +357,35 @@ export class TPCUtils {
         } else if (oldTrack && !newTrack) {
             return this.removeTrackMute(oldTrack)
                 .then(() => {
+                    const mediaType = oldTrack.getType();
+                    const transceiver = this._findTransceiver(mediaType);
+
+                    // Change the direction on the transceiver to 'recvonly' so that a 'removetrack'
+                    // is fired on the associated media stream on the remote peer.
+                    if (transceiver) {
+                        transceiver.direction = TransceiverDirection.RECVONLY;
+                    }
+
+                    // Remove the old track from the list of local tracks.
                     this.pc.localTracks.delete(oldTrack.rtcId);
                     this.pc.localSSRCs.delete(oldTrack.rtcId);
                 });
         } else if (newTrack && !oldTrack) {
             const ssrc = this.pc.localSSRCs.get(newTrack.rtcId);
 
             return this.addTrackUnmute(newTrack)
+                .then(() => this.setEncodings(newTrack))
```

Comment:
is `setEncodings` needed for the `if (oldTrack && newTrack)` case as well?

Reply:
It's not needed if there was a source already and only the stream is being replaced on the sender. However, if the client starts muted and the source is being added on unmute, then the encodings need to be configured.